File: daggermvp/src/main/java/be/vergauwen/simon/daggermvp/di/ApplicationModule.java
package be.vergauwen.simon.daggermvp.di;

import android.content.Context;
import be.vergauwen.simon.daggermvp.MVPApplication;
import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    private MVPApplication application;

    public ApplicationModule(MVPApplication application) {
        this.application = application;
    }

    @Provides
    @ApplicationScope
    public Context provideApplicationContext() {
        return application;
    }
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/di/ApplicationModule.kt
package be.vergauwen.simon.kotlindaggermvp.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val application: Application) {

  @ApplicationScope
  @Provides
  fun provideApplicationContext(): Context {
    return application
  }
}